### PR TITLE
resolves #189 reuse existing clone if available

### LINKED ISSUE #189: Only clone GitHub Pages repo if not already cloned
For some sites, the GitHub Pages repository is large and takes considerable time to clone (on top of the time it takes to first delete it from disk). Currently, the prepareGhPages task removes the existing folder, then clones the repository anew each time. It would save a lot of time if this plugin only cloned the repository if not already cloned. 

Ideally, the prepareGhPages task would check to see if the workingDir already exists and is a git repository. If both of those are true, the task would merely fetch updates.

### PR DESCRIPTION
**Pull request by mojavelinux**
- reuse existing clone if available & matches configuration
- fetch and hard reset clone before using
- only create orphan branch for fresh clone
- set/use ext.repo more efficiently

### CODE REVIEW

**Comment by mojavelinux** [comment]
The check for an existing repository is very strict. It must be the same origin url and the correct branch must be checked out. That's how we can be sure it's either the repository we used on a previous build or one pre-configured to be in the same state.

If we do find a repository we can use, we perform a fetch followed by a hard reset against the target branch. We aren't interested in any dirty changes. The whole point of this exercise is to save time by not unnecessarily cloning a large repository.

File: src/main/groovy/org/ajoberstar/gradle/git/ghpages/GithubPagesPlugin.groovy [hunk 1]
```diff
@@ -56,28 +59,49 @@ class GithubPagesPlugin implements Plugin<Project> {
...
-						credentials: extension.credentials?.toGrgit()
-				)
+				def repo = null
+				try {
+					// attempt to reuse existing repository
+					repo = Grgit.open(dir: extension.workingDir)
+					if (extension.repoUri == repo.remote.list().find { it.name == 'origin' }?.url &&
+							repo.branch.current.name == extension.targetBranch) {
+						repo.fetch()
+						repo.reset(commit: 'origin/' + extension.targetBranch, mode: ResetOp.Mode.HARD)
```
Comment (ajoberstar) [inline src/main/groovy/org/ajoberstar/gradle/git/ghpages/GithubPagesPlugin.groovy]: We should probably also run [`repo.clean(directories: true, ignore: false)`](http://ajoberstar.org/grgit/docs/groovydoc/org/ajoberstar/grgit/operation/CleanOp.html) to get rid of any untracked files.

**Comment by ajoberstar** [comment]
This looks good to me overall. I added one comment about some additional cleaning. I'll merge this in once I start working on 1.6.0 (or once I get 1.5.1 out). Thanks!

**Comment by mojavelinux** [comment]
PR updated!

**Comment by ajoberstar** [comment]
Thanks again!

**Pull request merged**